### PR TITLE
ci: Include documentationUri in module index data

### PR DESCRIPTION
**Pull request by shenglol**
This is the first step to fix https://github.com/Azure/bicep/issues/12122.

### CODE REVIEW

File: scripts/github-actions/generate-module-index-data.js [hunk 1]
```diff
@@ -68,157 +45,107 @@ async function getModuleDescription(
...
-            context
-          );
-          if (description) {
-            properties[tag] = { description };
-            modulesWithDescriptions[modulePath] = true;
+  for (const avmModuleRoot of ["avm/res", "avm/ptn"]) {
+    const avmModuleGroups = await getSubdirNames(fs, avmModuleRoot);
+
+    for (const moduleGroup of avmModuleGroups) {
+      const moduleGroupPath = `avm/res/${moduleGroup}`;
```
Comment (jtracey93) [inline scripts/github-actions/generate-module-index-data.js:105]: Think we need some logic here to handle the pattern modules here too.

As in the 2 separate loops we had before handling this had the following values for:

- Resource module ```const modulePath = `avm/res/${moduleGroup}/${moduleName}`;```
- Pattern module ```const modulePath = `avm/ptn/${moduleName}`;```

**Review by jtracey93 — CHANGES_REQUESTED** [review]
Hey @shenglol, just one thing about pattern modules i think we need to handle, but looks good otherwise 👍

File: scripts/github-actions/generate-module-index-data.js (hunk 1 above)
Comment (shenglol) [reply scripts/github-actions/generate-module-index-data.js:105]: Good catch...I meant to use `avmModuleGroup` here. Will fix it.

Comment (shenglol) [reply scripts/github-actions/generate-module-index-data.js]: Done.

**Review by jtracey93 — APPROVED** [review]
LGTM 🚀

**Pull request merged**